src/expr/impl/src/scalar/ai_model.rs

```diff
@@ -35,13 +36,45 @@ pub struct OpenAiEmbeddingContext {
 
 impl OpenAiEmbeddingContext {
     /// Create a new `OpenAI` embedding context from `api_key` and model
-    pub fn from_config(api_key: &str, model: &str) -> Result<Self> {
-        let config = OpenAIConfig::new().with_api_key(api_key);
+    pub fn from_config(config: Value) -> Result<Self> {
+        let get_string = |key: &str| -> Result<Option<String>> {
+            config
+                .get(key)
+                .map(|value| {
+                    value
+                        .as_str()
+                        .map(Into::into)
+                        .ok_or_else(|| invalid_param_err(format!("`{}` must be string", key)))
+                })
+                .transpose()
+        };
+
+        let expect_string = |key: &str| -> Result<String> {
+            get_string(key)?.ok_or_else(|| invalid_param_err(format!("`{}` must be set", key)))
+        };
+
+        let model = expect_string("model")?;
+        let api_key = get_string("api-key")?;
+        let org_id = get_string("org-id")?;
+        let proj_id = get_string("project-id")?;
+        let api_base = get_string("api-base")?;
```

Comment:
Wondering if it's some convention to use `kebab-case` as the configuration key here?

Reply:
No, it's just a random choice. Any suggestion?

Reply:
After applying https://github.com/risingwavelabs/risingwave/pull/22960#discussion_r2300354879, it is easier to use camel style.